test/test.js

```diff
@@ -1205,6 +1212,15 @@ describe('Page', function() {
       expect(response.status()).toBe(200);
       expect(response.url()).toContain('self-request.html');
     });
+    // TODO edit to vefify that results are retrieved
+    it('should support ignoreHTTPSErrors option', async({page_https, httpsServer}) => {
+      let error = null;
+      const response = await page_https.goto(httpsServer.EMPTY_PAGE).catch(e => error = e);
+      expect(error).toBe(null);
+      expect(response.ok()).toBe(true);
+      expect(response.securityDetails()).toBeTruthy();
+      expect(response.securityDetails().protocol).toBe('TLS 1.2');
```

Comment:
let's make SecurityDetails class and convert all fields to methods.

We don't have any getters/properties anywhere in the API.